core/nudge.c

```diff
@@ -438,6 +442,23 @@ handle_nudge(dcontext_t *dcontext, nudge_arg_t *arg)
         detach_helper(DETACH_NORMAL_TYPE);
     }
 #endif
+#ifdef LINUX
+    /* The detach handler is last since in the common case it doesn't return. */
+    if (TEST(NUDGE_GENERIC(detach), nudge_action_mask)) {
+        nudge_action_mask &= ~NUDGE_GENERIC(detach);
+        heap_error_code_t error_code_reserve, error_code_commit;
+        void *d_r_detachstack =
+            os_heap_reserve(NULL, DYNAMORIO_STACK_SIZE, &error_code_reserve, false);
+        if (!os_heap_commit(d_r_detachstack, DYNAMORIO_STACK_SIZE,
```

Comment:
This memory is not freed, right?  Add a XXX comment about that, that we leak it.

Reply:
done